ktor-server/ktor-server-core/jvm/src/io/ktor/server/http/content/StaticContent.kt

```diff
@@ -33,6 +33,7 @@ private val StaticContentAutoHead = createRouteScopedPlugin("StaticContentAutoHe
 
     on(ResponseBodyReadyForSend) { call, content ->
         check(call.request.local.method == HttpMethod.Head)
+        if (content is OutgoingContent.ReadChannelContent) content.readFrom().cancel(null)
```

Comment:
Why is it only for `ReadChannelContent`, shouldn't it also be for others `OutgoingContent` such as `ByteArrayContent`?

Reply:
Other types of content do not hold resources that need to be closed.